Neos.Media/Classes/Domain/Model/AssetSource/Neos/NeosAssetProxyInterface.php

```diff
@@ -0,0 +1,189 @@
+<?php
+namespace Neos\Media\Domain\Model\AssetSource\Neos;
+
+/*
+ * This file is part of the Neos.Media package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+  *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations\Inject;
+use Neos\Flow\Http\Uri;
+use Neos\Flow\Persistence\PersistenceManagerInterface;
+use Neos\Flow\ResourceManagement\ResourceManager;
+use Neos\Media\Domain\Model\Asset;
+use Neos\Media\Domain\Model\AssetInterface;
+use Neos\Media\Domain\Model\AssetSource\AssetProxy\AssetProxyInterface;
+use Neos\Media\Domain\Model\AssetSource\AssetSourceInterface;
+use Neos\Media\Domain\Model\ImageInterface;
+use Neos\Media\Exception\AssetServiceException;
+use Neos\Media\Exception\ThumbnailServiceException;
+use Psr\Http\Message\UriInterface;
+
+final class NeosAssetProxyInterface implements AssetProxyInterface
+{
+    /**
+     * @var NeosAssetSource
+     */
+    private $assetSource;
+
+    /**
+     * @var Asset
+     */
+    private $asset;
+
+    /**
+     * @Inject()
```

Comment:
Mhm, interesting variant to only use the Inject annotation. But inconsistent to the other classes, less expressive and how we show it in the documentations. Are there any benefits for that?

Reply:
I like this one better and started using it in other projects, but consistency rules!

Reply:
How can this even work?

Reply:
@kitsunet well, it does work, that's what the Doctrine annotation driver provides. Actually, importing a whole (sub-) namespace, like we do with "Flow" is sort of a exotic case.